lib/wallet/mod.rs

```diff
@@ -1931,4 +1931,280 @@ impl Wallet {
     ) -> Result<(), miette::Report> {
         self.inner.create_new_wallet(mnemonic, password).await
     }
+
+    /// Create a multisig address requiring multiple signatures to spend from.
+    ///
+    /// # Arguments
+    /// * `required_signatures` - Number of signatures required (m in m-of-n)
+    /// * `public_keys` - List of public keys in hex format
+    /// * `address_type` - Type of address to create: "p2sh", "p2wsh", or "p2sh-p2wsh"
+    ///
+    /// # Returns
+    /// Returns a tuple containing:
+    /// * The multisig address
+    /// * The redeem script in hex
+    /// * The output descriptor
+    pub fn create_multisig_address(
+        &self,
+        required_signatures: u32,
+        public_keys: &[String],
+        address_type: Option<&str>,
+    ) -> Result<(bitcoin::Address, String, String)> {
+        // Validate parameters
+        if required_signatures == 0 {
+            return Err(miette!("required_signatures must be greater than 0"));
+        }
+
+        if required_signatures as usize > public_keys.len() {
+            return Err(miette!(
+                "required_signatures ({}) cannot be greater than the number of public keys ({})",
+                required_signatures,
+                public_keys.len()
+            ));
+        }
+
+        if public_keys.is_empty() {
+            return Err(miette!("public_keys cannot be empty"));
+        }
```

Comment:
This should be validated in `server.rs` IMO, in a way that proper tonic statuses with the `InvalidArgument` status code is returned.